core/src/main/java/org/apache/brooklyn/core/OsgiLauncher.java

```diff
@@ -0,0 +1,557 @@
+//        } else {
+//            if (!permission.get().subSequence(4, 10).equals("------")) {
+//                throw new FatalRuntimeException("Invalid permissions for file " + f + "; expected ?00 but was " + permission.get());
+//            }
+//        }
+//    }
+
+    /* equivalent of launcher.start() in Main.java */
+    private void launcherStart() {
+        // not needed: management context already running (courtesy of blueprint.xml)
+//        // Create the management context
+//        initManagementContext();
+
+        // Inform catalog initialization that it is starting up
+        CatalogInitialization catInit = ((ManagementContextInternal) managementContext).getCatalogInitialization();
+        catInit.setStartingUp(true);
+
+        // not needed: webapps are started as different features (brooklyn-rest-resources-cxf, brooklyn-jsgui etc)
+//        // Start webapps as soon as mgmt context available -- can use them to detect progress of other processes
+//        if (startWebApps) {
+//            try {
+//                startWebApps();
+//            } catch (Exception e) {
+//                handleSubsystemStartupError(ignoreWebErrors, "core web apps", e);
+//            }
+//        }
+        // not needed: camp is a separate karaf feature - let it initialize itself
+//        // Add a CAMP platform
+//        campPlatform = new BrooklynCampPlatformLauncherNoServer()
+//                .useManagementContext(managementContext)
+//                .launch()
+//                .getCampPlatform();
+        try {
+            initPersistence();
+            startPersistence();
+        } catch (Exception e) {
+            handleSubsystemStartupError(ignorePersistenceErrors, "persistence", e);
+        }
+
+        try {
+            // run cat init now if it hasn't yet been run;
+            // will also run if there was an ignored error in catalog above, allowing it to fail startup here if requested
+            if (!catInit.hasRunOfficialInitialization()) {
+                if (persistMode == PersistMode.DISABLED) {
+                    log.debug("Loading catalog as part of launch sequence (it was not loaded as part of any rebind sequence)");
+                    catInit.populateCatalog(ManagementNodeState.MASTER, true, true, null);
+                } else {
+                    // should have loaded during rebind
+                    ManagementNodeState state = managementContext.getHighAvailabilityManager().getNodeState();
+                    log.warn("Loading catalog for " + state + " as part of launch sequence (it was not loaded as part of the rebind sequence)");
+                    catInit.populateCatalog(state, true, true, null);
+                }
+            }
+        } catch (Exception e) {
+            handleSubsystemStartupError(ignoreCatalogErrors, "initial catalog", e);
+        }
+        catInit.setStartingUp(false);
+
+        // no need for command-line locations
```

Comment:
Personally I would not use command-line arguments within karaf, so I have not researched this. But I think it can be technically done if so wanted.

Here is why I want to stay away from command-line args:
- bundle lifecycle is different from karaf lifecycle. How do we handle command-line arguments if we restart `brooklyn-core`? Do we assume the previous life of `brookyn-core` handled those and we don't act on them? Or should they be re-applied? If so, we need to store this somehwere, so why not set our wishes in the config files from the beginning?
- how do we handle command-line arguments on different karaf cellar nodes?
- karaf has a command-line shell where we can place commands for users to play with. I think karaf defines a startup script (i.e. like bashrc) and this could be a place to issue startup commands